aiida_sssp_workflow/workflows/convergence/base.py

```diff
@@ -0,0 +1,346 @@
+# -*- coding: utf-8 -*-
+"""
+The base convergence WorkChain
+"""
+from abc import ABCMeta, abstractmethod
+import typing as ty
+
+from aiida_tools.process_inputs import get_fullname
+
+from aiida.engine import WorkChain, ToContext
+from aiida import orm
+from aiida.plugins import WorkflowFactory
+
+from aiida_sssp_workflow.workflows.convergence.engine import TwoInputsTwoFactorsConvergence
+from ..helper import get_pw_inputs_from_pseudo
+
+CreateEvaluateWorkChain = WorkflowFactory('optimize.wrappers.create_evaluate')
+OptimizationWorkChain = WorkflowFactory('optimize.optimize')
+
+PARA_ECUTWFC_LIST = lambda: orm.List(list=[
```

Comment:
Yes, I use it as the default value of WorkChain so better to be a lambda.

Reply:
Yes, but it seems like you use `lambda` twice: once when you define the constant, and again when you define the default value.

Reply:
hehe, that's true. :-p

Reply:
Just a quick note, because I just noticed how easy it is to get this wrong: Here it's important to keep in mind why we need the `lambda`, which is to ensure that each new process gets a new AiiDA node as input.

So the following:
```python
PARA_ECUTRHO_LIST = orm.List(list=[...])

spec.input(
    ...,
    default=lambda: PARA_ECUTWFC_LIST,
)
```
would _not_ solve the problem, because the lambda always returns the same node.

This would work on the other hand:
```python
PARA_ECUTRHO_LIST = lambda: orm.List(list=[...])

spec.input(
    ...,
    default=PARA_ECUTWFC_LIST,
)
```

or even
```python
PARA_ECUTRHO_LIST = [...]

spec.input(
    ...,
    default=lambda: orm.List(list=PARA_ECUTRHO_LIST),
)
```